src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -11,13 +10,27 @@
 
 from ...basedatainterface import BaseDataInterface
 from ...tools.spikeinterface import write_recording
-from ...utils import get_schema_from_hdmf_class, get_base_schema, OptionalFilePathType
+from ...utils import get_schema_from_hdmf_class, get_base_schema, OptionalFilePathType, get_package
 
 
-class BaseRecordingExtractorInterface(BaseDataInterface, ABC):
-    """Primary class for all RecordingExtractorInterfaces."""
+class _LazyRecordingExtractor(type(BaseDataInterface), type):
+    def __getattribute__(self, name):
+        if name == "RX" and super().__getattribute__("RX") is None:
+            extractor_module = get_package(package_name=self.RXModule or "spikeinterface.extractors")
+            return getattr(extractor_module, self.RXName or self.__name__.replace("Interface", "Extractor"))
+        return super().__getattribute__(name)
 
-    RX = None
+
+class BaseRecordingExtractorInterface(BaseDataInterface, metaclass=_LazyRecordingExtractor):
+    """Parent class for all RecordingExtractorInterfaces."""
+
+    RXModule: Optional[str] = None  # Defaults to "spikeinterface.extractors". Manually override in subclass if needed.
+    RXName: Optional[str] = None  # Defaults to __name__.replace("Interface", "Extractor"). Manually override if needed.
+    RX = None  # Loads dynamically on first access attempt
+
+    def __new__(cls, *args, **kwargs):
+        cls.RX = getattr(cls, "RX")
+        return object.__new__(cls)
```

Comment:
This code structure repeats with slight differences over each extractor interface. One way to do some code reduction would be to define a `BaseExtractorInterface` with a general attribute like `ExtractorClass` instead of specific `RX/SegX/IX/SX`